cdap-data-fabric/src/test/java/co/cask/cdap/spi/metadata/MetadataStorageTest.java

```diff
@@ -344,6 +344,70 @@ public void testUpdateRemove() throws IOException {
+    MetadataRecord app1Record = new MetadataRecord(app1, new Metadata(tags(), props(descSystem,
+                                                                                    "this is the first application",
+                                                                                    descUser,
+                                                                                    "business description of app1")));
+    MetadataRecord app2Record = new MetadataRecord(app2, new Metadata(tags(), props(descSystem,
+                                                                                    "this other app description")));
+    // add some metadata with descriptions
+    mds.batch(batch(new Update(app1, app1Record.getMetadata()),
+                    new Update(app2, app2Record.getMetadata())));
+
+    // search with field description:
+    assertResults(mds, SearchRequest.of("description:first").build(), app1Record);
+    assertResults(mds, SearchRequest.of("description:this").build(), app1Record, app2Record);
+    assertResults(mds, SearchRequest.of("description:app*").build(), app1Record, app2Record);
+    assertResults(mds, SearchRequest.of("description:description").build(), app1Record, app2Record);
+
+    // search in SCOPE SYSTEM
+    assertResults(mds, SearchRequest.of("description:first").setScope(SYSTEM).build(), app1Record);
+    assertResults(mds, SearchRequest.of("description:this").setScope(SYSTEM).build(), app1Record, app2Record);
+    assertResults(mds, SearchRequest.of("description:app*").setScope(SYSTEM).build(), app1Record, app2Record);
+    assertResults(mds, SearchRequest.of("description:description").setScope(SYSTEM).build(), app2Record);
+
+    // search in scope USER
+    assertEmpty(mds, SearchRequest.of("description:first").setScope(USER).build());
+    assertEmpty(mds, SearchRequest.of("description:this").setScope(USER).build());
+    assertResults(mds, SearchRequest.of("description:app*").setScope(USER).build(), app1Record);
+    assertResults(mds, SearchRequest.of("description:description").setScope(USER).build(), app1Record);
+
+    // search plain text should match description text but also the field name "description"
+    assertResults(mds, SearchRequest.of("first").build(), app1Record);
+    assertResults(mds, SearchRequest.of("this").build(), app1Record, app2Record);
+    assertResults(mds, SearchRequest.of("app*").build(), app1Record, app2Record);
+    assertResults(mds, SearchRequest.of("description").build(), app1Record, app2Record);
+
+    // search in SCOPE SYSTEM
+    assertResults(mds, SearchRequest.of("first").setScope(SYSTEM).build(), app1Record);
+    assertResults(mds, SearchRequest.of("this").setScope(SYSTEM).build(), app1Record, app2Record);
+    assertResults(mds, SearchRequest.of("app*").setScope(SYSTEM).build(), app1Record, app2Record);
+    assertResults(mds, SearchRequest.of("description").setScope(SYSTEM).build(), app2Record);
```

Comment:
Since plain text matches field name "description", this query should return `app1Record` too, right?

Reply:
app1 only matches in scope user, but this searches in scope system

Reply:
Maybe I misunderstood the comment `search plain text should match description text but also the field name "description"` then. 
Since the comment indicated that the field name "description" also matches, I expected  `app1Record` to be returned since it has "description" named field in system scope - `descSystem`. What does the comment mean?

Reply:
ehm, I need to update comment, it is inaccurate. 

There is a follow-up PR that introduces searching by property name.

Reply:
I see